cmake/std/atdm/ats2/environment.sh

```diff
@@ -0,0 +1,231 @@
+#
+################################################################################
+
+# ats2 jobs all use the same environmnet changes to the
+# sourced script below will impact jobs on both of those
+# machines. please be mindful of this when making changes
+
+# Handle KOKKOS_ARCH
+
+if [[ "$ATDM_CONFIG_COMPILER" == "GNU"* || \
+  "$ATDM_CONFIG_COMPILER" == "XL"* ]]; then
+  if [[ "$ATDM_CONFIG_KOKKOS_ARCH" == "DEFAULT" || \
+    "$ATDM_CONFIG_KOKKOS_ARCH" == "Power9" ]] ; then
+    export ATDM_CONFIG_KOKKOS_ARCH=Power9
+    arch=pwr9
+  else
+    echo
+    echo "***"
+    echo "*** ERROR: KOKKOS_ARCH=$ATDM_CONFIG_KOKKOS_ARCH is not a valid option"
+    echo "*** for the compiler GNU.  Replace '$ATDM_CONFIG_KOKKOS_ARCH' in the"
+    echo "*** job name with 'Power9'"
+    echo "***"
+    return
+  fi
+elif [[ "$ATDM_CONFIG_COMPILER" == "CUDA"* ]] ; then
+  if [[ "$ATDM_CONFIG_KOKKOS_ARCH" == "DEFAULT" || \
+    "$ATDM_CONFIG_KOKKOS_ARCH" == "Power9" || \
+    "$ATDM_CONFIG_KOKKOS_ARCH" == "Volta70" ]] ; then
+    export ATDM_CONFIG_KOKKOS_ARCH=Volta70
+    arch=v100
+  else
+    echo
+    echo "***"
+    echo "*** ERROR: KOKKOS_ARCH=$ATDM_CONFIG_KOKKOS_ARCH is not a valid option"
+    echo "*** for the CUDA compiler.  Replace '$ATDM_CONFIG_KOKKOS_ARCH' in the"
+    echo "*** job name with one of the following options:"
+    echo "***"
+    echo "***   'Volta70' Power9 with Volta V-100 GPU (Default)"
+    echo "***"
+    return
+  fi
+else
+  echo
+  echo "***"
+  echo "*** ERROR: COMPILER=$ATDM_CONFIG_COMPILER is not supported on this system!"
+  echo "***"
+  return
+fi
+
+export ATDM_CONFIG_SPARC_TPL_BASE=/projects/sparc/tpls/ats2-${arch}
+
+echo "Using $ATDM_CONFIG_SYSTEM_NAME toss3 compiler stack $ATDM_CONFIG_COMPILER to build $ATDM_CONFIG_BUILD_TYPE code with Kokkos node type $ATDM_CONFIG_NODE_TYPE"
+
+export ATDM_CONFIG_ENABLE_SPARC_SETTINGS=ON
+export ATDM_CONFIG_USE_NINJA=OFF
+export ATDM_CONFIG_BUILD_COUNT=8
+# export ATDM_CONFIG_CMAKE_JOB_POOL_LINK=2
+# NOTE: Above, currently setting CMAKE_JOB_POOL_LINK results in a build
+# failures with Ninja.  See https://gitlab.kitware.com/snl/project-1/issues/60
+
+if [ "$ATDM_CONFIG_NODE_TYPE" == "OPENMP" ] ; then
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=8
+  export OMP_NUM_THREADS=2
+else
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=16
+fi
+
+# Common modules for all builds
+module load git/2.20.0
+module load cmake/3.14.5
+
+if [[ "$ATDM_CONFIG_COMPILER" == *"GNU-7.3.1_SPMPI-2019.06.24"* ]]; then
+  module load gcc/7.3.1
+  module load lapack/3.8.0-gcc-4.9.3
+
+  # rabartl: ToDo: Above, we need to find a way to extract 'ats2-pwr9' out of
+  # this file for this to be general!
+
+  export CBLAS_ROOT=/usr/tcetmp/packages/lapack/lapack-3.8.0-gcc-4.9.3
+  export LAPACK_ROOT=/usr/tcetmp/packages/lapack/lapack-3.8.0-gcc-4.9.3
+  export COMPILER_ROOT=/usr/tce/packages/gcc/gcc-7.3.1
+  export SPARC_HDF5=hdf5-1.10.5
+
+  # eharvey: TODO: remove COMPILER_ROOT and other unused exports below.
```

Comment:
Does COMPILER_ROOT really need to be removed?

Reply:
I haven't got to making this succinct yet. I don't think we need *all* of these environment variables set.